src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/common/BerConverter.cs

```diff
@@ -454,12 +454,13 @@ private static int EncodingByteArrayHelper(BerSafeHandle berElement, byte[] temp
                 IntPtr tmp = Marshal.AllocHGlobal(tempValue.Length);
                 Marshal.Copy(tempValue, 0, tmp, tempValue.Length);
                 HGlobalMemHandle memHandle = new HGlobalMemHandle(tmp);
-
-                error = Wldap32.ber_printf_bytearray(berElement, new string(fmt, 1), memHandle, tempValue.Length);
+                error = NativePal.BerPrintfBytearray(berElement, new string(fmt, 1), memHandle, tempValue.Length);
             }
             else
             {
-                error = Wldap32.ber_printf_bytearray(berElement, new string(fmt, 1), new HGlobalMemHandle(IntPtr.Zero), 0);
+                IntPtr tmp = Marshal.AllocHGlobal(0);
```

Comment:
>0 [](start = 50, length = 1)

This looks strange :-) allocating 0 bytes and wrapping it inside safehandle. does the native call can grow the memory?

Reply:
yeah I agree but if I try calling the native api with a handle of IntPtr.Zero, the native library throws an error. If instead y actually create a pointer like this it works.

Reply:
Either you can get a pointer on the stack (using stackalloc and then wraps it inside IntPtr) or you can have a single static instance you can use it all the time.